include/cutlass/gemm/collective/builders/pvc_dpas_builder.inl

```diff
@@ -0,0 +1,128 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ **************************************************************************************************/
+
+#pragma once
+
+#include <cutlass/arch/arch.h>
+#include <cutlass/gemm/dispatch_policy.hpp>
+
+#include "cutlass/gemm/collective/collective_mma.hpp"
+
+namespace cutlass::gemm::collective {
+
+  // Intel PVC Single stage dpas pipeline
+  // Also the auto builder
+
+template <
+  class ElementA,
+  class GmemLayoutATag,
+  int AlignmentA,
+  class ElementB,
+  class GmemLayoutBTag,
+  int AlignmentB,
+  class ElementAccumulator,
+  class TileShape_MNK,
+  class KernelScheduleType
+  > 
+struct CollectiveBuilder<
+  arch::IntelPVC,
+  arch::OpClassTensorOp,   // Reusing opClassTensorOp instead of declaring a new "OpClassDpasOP" as they have the same meaning
+  ElementA,
+  GmemLayoutATag,
+  AlignmentA,
+  ElementB,
+  GmemLayoutBTag,
+  AlignmentB,
+  ElementAccumulator,
+  TileShape_MNK,
+  Shape<_1, _1, _1>,    // Cluster Shape, which has no meaning in PVC
+  cutlass::gemm::collective::StageCountAuto, // Stage Count needn't be a parameter
+  KernelScheduleType,
+  cute::enable_if_t<
+    (cute::is_same_v<KernelScheduleType, KernelSinglestage> ||
+     cute::is_same_v<KernelScheduleType, KernelScheduleAuto>) &&  
+    cute::is_same_v<GmemLayoutATag, cutlass::layout::RowMajor> && // Different struct specialization because this will change copy atoms
+    cute::is_same_v<GmemLayoutBTag, cutlass::layout::RowMajor>
+  >
+    >{
+      #ifndef SYCL_INTEL_TARGET
```

Comment:
It can still be SYCL_INTEL_TARGET but the device might not be PVC

Reply:
we only define SYCL_INTEL_TARGET if the target is intel_gpu_pvc at the moment
so if it's SYCL_INTEL_TARGET, then it's definitely PVC

Reply:
Not necessarily,  We dont have that assumption. There are common examples that runs on other intel devices like ARC and this flag is getting used. So the INTEL_SYCL_TARGET is used for all Intel devices.

Reply:
We are setting it here - https://github.com/codeplaysoftware/cutlass-fork/blob/a10d18a58ebeb37f01921d98b7918032af1346f5/CMakeLists.txt#L103

Reply:
Again, this is target triple for DPCPP not Intel_SYCL_Target
we can still define the target and set the intel_sycl_target on without going through the automatic setting via CMAKE

Reply:
> Again, this is target triple for DPCPP not Intel_SYCL_Target

We set `SYCL_INTEL_TARGET` only if the dpcpp target is pvc, I did not realize the it was not the link to the subsequent lines as well, which are as follows

```cmake
  elseif(DPCPP_SYCL_TARGET STREQUAL "intel_gpu_pvc")
    set(SYCL_INTEL_TARGET ON)
    add_compile_definitions(SYCL_INTEL_TARGET)
  endif()
```

But I do get your point being that we want to use SYCL_INTEL_TARGET to cover a broader range of intel GPUs, and not just PVC, I have removed that check, though since we only support PVC at the moment, that's why SYCL_INTEL_TARGET and the "intel_gpu_pvc" are so tightly coupled. We might have to change the cmake accordingly later when we add support for more architectures.

Reply:
The check for the architecture is generic and should be removed from CMake. We will handle it in a separate PR. it should be ARCH based and based on that we would enable/disable different flags